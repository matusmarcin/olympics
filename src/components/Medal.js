import React, { PropTypes } from 'react';

const medalIcon = {
  gold: '🥇',
  silver: '🥈',
  bronze: '🥉',
};

const Medal = ({ athlete, event, medal, sex }) => (
  <li>
    {sex === 'Men' ? '🏃' : '🏃‍♀️'}
    {athlete} - {event} - <span>{medalIcon[medal.toLowerCase()]}</span>
  </li>
);

Medal.propTypes = {
  athlete: PropTypes.string.isRequired,
  sex: PropTypes.string.isRequired,
  event: PropTypes.string.isRequired,
  medal: PropTypes.string.isRequired,
};

export default Medal;
