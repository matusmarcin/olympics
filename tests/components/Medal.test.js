/* eslint-env jest, browser */
import React from 'react';
import renderer from 'react-test-renderer';
import Medal from '../../src/components/Medal';

it('renders correctly', () => {
  const tree = renderer.create(
    <Medal athlete="Matus Marcin" medal="Gold" event="test writing" sex="Men" />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly female', () => {
  const tree = renderer.create(
    <Medal athlete="Some Lady" medal="Gold" event="olympic games attendance" sex="Women" />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
