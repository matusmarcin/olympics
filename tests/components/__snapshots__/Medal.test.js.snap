exports[`test renders correctly 1`] = `
<li>
  🏃
  Matus Marcin
   - 
  test writing
   - 
  <span>
    🥇
  </span>
</li>
`;

exports[`test renders correctly female 1`] = `
<li>
  🏃‍♀️
  Some Lady
   - 
  olympic games attendance
   - 
  <span>
    🥇
  </span>
</li>
`;
